### PR TITLE
sys-cluster/teleport: add v2.3.5, deprecate v2.2.0 v2.2.1

### PR DESCRIPTION
**Pull request by gclawes**
- Add new teleport release v2.3.0
  - NOTE: The 2.3.0 ebuild re-enables pre-stripping binaries.  The strip step in portage removes the appended webassets data from the `teleport` binary, preventing web UI from working.  See [gentoo bug #631076](https://bugs.gentoo.org/show_bug.cgi?id=631076) for more details.  If there is a better way to preserve the appended data I can add that to this PR.
- Deprecate old versions v2.2.0, v2.2.1

### CODE REVIEW

File: sys-cluster/teleport/teleport-2.3.0.ebuild [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+else
+	inherit golang-vcs-snapshot
+	SRC_URI="https://github.com/gravitational/${PN}/archive/v${PV}.tar.gz -> ${P}.tar.gz"
+	KEYWORDS="~amd64 ~arm"
+fi
+
+LICENSE="Apache-2.0"
+SLOT="0"
+IUSE=""
+RESTRICT="test"
```
Comment (mgorny) [inline sys-cluster/teleport/teleport-2.3.0.ebuild]: You probably want explicit `RESTRICT=strip` here.

Comment (gclawes) [reply sys-cluster/teleport/teleport-2.3.0.ebuild]: Done.

**Comment by gclawes** [comment]
Bumped v2.3.0 -> v2.3.5, enabled RESTRICT=strip

**Comment by gclawes** [comment]
Closing, v2.4.0 has been released, will open up new PR.

**Pull request closed without merging**